api/plugins/action/deploy_bulk.py

```diff
@@ -0,0 +1,137 @@
+    lagoon.api.deploy_bulk:
+    name: Trigger by Ansible
+    environments:
+        - name: environment_name
+        project:
+            name: project_name
+    build_vars:
+        - name: build_var_name
+        value: build_var_value
+'''
+
+display = Display()
+
+
+def deploy_bulk(client: GqlClient, build_vars: list, name: str, envs: list) -> dict:
+
+    res = client.execute_query(
+        """
+        mutation bulkDeployEnvironment(
+            $build_vars:[EnvKeyValueInput]
+            $envs: [DeployEnvironmentLatestInput!]!
+            $name: String
+        ) {
+            bulkDeployEnvironmentLatest(input: {
+                name: $name
+                buildVariables: $build_vars
+                environments: $envs
+            })
+        }""",
+        {
+            "build_vars": build_vars,
+            "name": name,
+            "envs": envs,
+        }
+    )
+
+    display.v(f"GraphQL query result: {res}")
+    return res['bulkDeployEnvironmentLatest']
+
+
+def is_variable_type(i):
+    if type(i) is not dict:
+        return False, 'Expected "dict" type'
+    keys = i.keys()
+
+    if 'name' not in keys and 'value' not in keys:
+        return False, 'Required keys "name" and "value" missing'
+
+    return True, None
+
+def is_environment_type(i):
+    if type(i) is not dict:
+        return False, 'Expected "dict" type'
+
+    keys = i.keys()
+
+    if 'project' not in keys and 'id' not in keys:
+        return False, 'Required keys "project" or "id" missing'
+
+    if 'project' in keys and 'id' in keys:
+        return False, 'Please specify one "project" or "id" for environment input'
+
+    if 'project' in keys:
+        project_keys = i['project'].keys()
+
+        if 'name' not in keys:
+            return False, 'Project type requires environment "name"'
+
+        if 'name' not in project_keys and 'id' not in project_keys:
+            return False, 'Required keys "name" or "id" for project'
+
+    return True, None
+
+
+class ActionModule(ActionBase):
+
+    def run(self, tmp=None, task_vars=None):
+
+        if task_vars is None:
+            task_vars = dict()
+
+        result = super(ActionModule, self).run(tmp, task_vars)
+
+        lagoon = GqlClient(
+            self._templar.template(task_vars.get('lagoon_api_endpoint')).strip(),
+            self._templar.template(task_vars.get('lagoon_api_token')).strip(),
+            self._task.args.get('headers', {})
+        )
+
+        b = self._task.args.get('build_vars')
+        n = self._task.args.get('name')
+        e = self._task.args.get('environments')
+        envs = []
+
+        result['invalid_variable'] = []
+        result['invalid_environment'] = []
+
+        for i in range(len(b)):
+            valid, r = is_variable_type(b[i])
+            if not valid:
+                del b[i]
+                result['invalid_variable'].append(b[i])
+                display.v(f'Invalid project detected: {r}')
+                display.v(b[i])
```

Comment:
Should `del b[i]` come at the end instead? Otherwise you're deleting it and then using it in the subsequent lines.

```suggestion
                result['invalid_variable'].append(b[i])
                display.v(f'Invalid project detected: {r}')
                display.v(b[i])
                del b[i]
```